include/navigation/image_database.h

```diff
@@ -73,78 +79,245 @@ class ImageDatabase
     //! The metadata for an entry in an ImageDatabase
     struct Entry
     {
-        Vector3<millimeter_t> position;
-        degree_t heading;
+        Vector3<millimeter_t> position = Vector3<millimeter_t>::nan();
+        degree_t heading{ NAN };
         filesystem::path path;
         std::array<size_t, 3> gridPosition; //! For grid-type databases, indicates the x,y,z grid position
+        std::unordered_map<std::string, std::string> extraFields;
```

Comment:
I'm in two minds on this one but, especially as the ``record`` API is based on adding elements in order, it might be a lot simpler to just make this a nested class and have:

```c++
std::array<std::string, NumExtraFields> extraFields;
```
Then, if you wanted to,  you could always make an application-side enum or whatever.

Reply:
This did occur to me after I'd submitted the PR... I think it's probably the more sensible way of doing it tbh. You could still have the helper function to get the value from its string name.

We probably shouldn't rely on fields being in a particular order though. I've deliberately made the CSV parsing code agnostic to column order now because there were a few hacks that made assumptions about the data based on number of columns and it's not v extensible that way.

Reply:
...although after fiddling with the code I now realise we'd need ``Entry``s to hold a reference to the ``ImageDatabase`` and then use that to figure out which field corresponds to a given key. So it's probably cleaner as is.